packages/taco-auth/src/providers/eip4361/eip4361.ts

```diff
@@ -61,6 +67,19 @@ export class EIP4361AuthProvider {
     return authMessage;
   }
 
+  private isMessageExpired(message: string): boolean {
+    const siweMessage = new SiweMessage(message);
+    if (!siweMessage.issuedAt) {
+      // don't expect to ever happen; but just in case
+      return false;
+    }
```

Comment:
This is a sentinel against `undefined`?

Reply:
Yes.

From the EIP4361 spec,`issued-at` is a required value - https://docs.login.xyz/general-information/siwe-overview/eip-4361#specification.

However, in the `siwe` typescript library implementation, it is an optional value in the constructor - https://github.com/spruceid/siwe/blob/main/packages/siwe/lib/client.ts#L49. As a result, Typescript complains if we don't check whether it is defined first before using it.

Reply:
Is this a divergence of `siwe` wrt to the spec then?

Reply:
As far as I can tell, yes, it is a divergence.

Reply:
Well, it's not the constructor but the SiweMessage struct - so still a deviation - https://github.com/spruceid/siwe/blob/main/packages/siwe/lib/client.ts#L49

Without the check, the compiler returns the following:
```
packages/taco-auth/src/providers/eip4361/eip4361.ts:72:36 - error TS2769: No overload matches this call.
  Overload 1 of 4, '(value: string | number | Date): Date', gave the following error.
    Argument of type 'string | undefined' is not assignable to parameter of type 'string | number | Date'.
      Type 'undefined' is not assignable to type 'string | number | Date'.
  Overload 2 of 4, '(value: string | number): Date', gave the following error.
    Argument of type 'string | undefined' is not assignable to parameter of type 'string | number'.
      Type 'undefined' is not assignable to type 'string | number'.
 ```
 
 I'll file an issue for the `siwe` library, and perhaps they can either clarify or fix.